numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2635,7 +2635,7 @@ static PyObject *
             Py_DECREF(typecode);
             return NULL;
         }
-        memset(&((Py@Name@ScalarObject *)robj)->obval, 0, sizeof(npy_@name@));
+        memset(&PyArrayScalar_VAL(obj, @Name@), 0, sizeof(npy_@name@));
```

Comment:
```suggestion
        memset(&PyArrayScalar_VAL(robj, @Name@), 0, sizeof(npy_@name@));
```
Probably this, maybe another one. I am good with this (not that I like that macro too much, but its a slight improvement).

Reply:
I had to look at this in my email client 3 times to spot the difference...

Reply:
Did the trick, thanks. Probably worth looking through for any others like this.

Reply:
Thanks, went through it, all looked good. Funny how we have this for Object, but it should not be possible to create Object scalars...

Reply:
I was thinking that too...